addons/GME/Scripts/Game/GME/GameMode/SCR_BaseGameMode.c

```diff
@@ -0,0 +1,57 @@
+//------------------------------------------------------------------------------------------------
+modded class SCR_BaseGameMode : BaseGameMode
+{
+	[Attribute(defvalue: "NONE", desc: "Global AI skill", uiwidget: UIWidgets.ComboBox, enums: ParamEnumArray.FromEnum(EAISkill))]
+	protected EAISkill m_eGME_GlobalAISkill;
+	
+	//------------------------------------------------------------------------------------------------
+	//! Load mission header settings
+	override protected void OnGameStart()
+	{
+		super.OnGameStart();
+		
+		SCR_MissionHeader header = SCR_MissionHeader.Cast(GetGame().GetMissionHeader());
+		if (header)
+			m_eGME_GlobalAISkill = header.m_eGME_GlobalAISkill;
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	//! Sets skill for all placed entities and future ones
+	void GME_SetGlobalAISkill(EAISkill skill)
+	{
+		m_eGME_GlobalAISkill = skill;
+		
+		if (skill == EAISkill.NONE)
+			return;
+		
+		SCR_EditableEntityCore core = SCR_EditableEntityCore.Cast(SCR_EditableEntityCore.GetInstance(SCR_EditableEntityCore));
+		if (!core)
+			return;
+		
+		set<SCR_EditableEntityComponent> entities = new set<SCR_EditableEntityComponent>();	
+		core.GetAllEntities(entities);
+		
+		foreach (SCR_EditableEntityComponent entity : entities)
+		{
+			SCR_EditableCharacterComponent editableChar = SCR_EditableCharacterComponent.Cast(entity);
+			if (!editableChar)
+				continue;
+		
+			IEntity char = editableChar.GetOwner();
+			if (!char)
+				continue;
+	
+			SCR_AICombatComponent combatComponent = SCR_AICombatComponent.Cast(char.FindComponent(SCR_AICombatComponent));
+			if (!combatComponent)
+				continue;
+		
+			combatComponent.SetAISkill(skill);
+		}
+	}
+	
+	//------------------------------------------------------------------------------------------------
+	EAISkill GME_GetGlobalAISkill()
+	{
+		return m_eGME_GlobalAISkill;
```

Comment:
Can we guarantee this variable will always be filled with the current AI Skill before this gets called? 

If ``onGameStart`` is called but there is no valid ``missionHeader`` then the initialized value of this value would not be called? 
However I don't know if it is possible to have a game where you got to the "zeus" screen without a valid mission header?

Reply:
All ingame scenarios have mission headers. The only case where there's no mission header is if you start a scenario via the workbench's play mode.

Reply:
> Can we guarantee this variable will always be filled with the current AI Skill before this gets called?

This is a valid question. I have to check how it behaves for workbench-placed units.

Reply:
Actually the question is more whether the global skill value gets applied for them. As for your actual question, the default value is `NONE` and as I wrote below, `NONE` means that the skill value does not get set for new AI and they have their default value in return.

Reply:
Good find. This value is in fact not correctly set for workbench placed AI. I fixed it in 4c18901 by using the setter instead, which also applies the correct value to already spawned AI.